internal/testgen/gen_suite_method_signature.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"text/template"
+
+	"github.com/Antonboom/testifylint/internal/checkers"
+)
+
+type SuiteMethodSignatureTestsGenerator struct{}
+
+func (SuiteMethodSignatureTestsGenerator) Checker() checkers.Checker {
+	return checkers.NewSuiteMethodSignature()
+}
+
+func (g SuiteMethodSignatureTestsGenerator) TemplateData() any {
+	name := g.Checker().Name()
+
+	return struct {
+		CheckerName               CheckerName
+		ReportMethodShouldBeClean string
+		ReportConflictWithIface   string
+	}{
+		CheckerName:               CheckerName(name),
+		ReportMethodShouldBeClean: QuoteReport(name + ": test method should not have any arguments or returning values"),
+		ReportConflictWithIface:   QuoteReport(name + ": method conflicts with suite.%s interface"),
+	}
+}
+
+func (SuiteMethodSignatureTestsGenerator) ErroredTemplate() Executor {
+	return template.Must(template.New("SuiteMethodSignatureTestsGenerator.ErroredTemplate").
+		Funcs(fm).
+		Parse(SuiteMethodSignatureTestTmpl))
+}
+
+func (SuiteMethodSignatureTestsGenerator) GoldenTemplate() Executor {
+	// NOTE(a.telyshev): Autofix may result in uncompiled code.
+	return nil
+}
+
+const SuiteMethodSignatureTestTmpl = header + `
+
+package {{ .CheckerName.AsPkgName }}
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type ValidSuite struct {
+	suite.Suite
+}
+
+func TestValidSuite(t *testing.T) {
+	suite.Run(t, new(ValidSuite))
+}
+
+func (s *ValidSuite) SetupSuite() {}
+func (s *ValidSuite) SetupTest() {}
+func (s *ValidSuite) TearDownSuite() {}
+func (s *ValidSuite) TearDownTest() {}
+func (s *ValidSuite) BeforeTest(suiteName, testName string) {}
+func (s *ValidSuite) AfterTest(suiteName, testName string) {}
+func (s *ValidSuite) HandleStats(suiteName string, stats *suite.SuiteInformation) {}
+func (s *ValidSuite) SetupSubTest() {}
+func (s *ValidSuite) TearDownSubTest() {}
+func (s *ValidSuite) TestTrue() { s.True(true) }
```

Comment:
Nitpicking:

I would add something like this

```suggestion
type ValidSuite struct {
	suite.Suite
}

func TestValidSuite(t *testing.T) {
	suite.Run(t, new(ValidSuite))
}

func (s *ValidSuite) SetupSuite() {}
func (s *ValidSuite) SetupTest() {}
func (s *ValidSuite) TearDownSuite() {}
func (s *ValidSuite) TearDownTest() {}
func (s *ValidSuite) BeforeTest(suiteName, testName string) {}
func (s *ValidSuite) AfterTest(suiteName, testName string) {}
func (s *ValidSuite) HandleStats(suiteName string, stats *suite.SuiteInformation) {}
func (s *ValidSuite) SetupSubTest() {}
func (s *ValidSuite) TearDownSubTest() {}
func (s *ValidSuite) TestTrue() { s.True(true) }

// interfaces validation
var _ suite.SetupAllSuite = ValidSuite{}
var _ suite.SetupTestSuite = ValidSuite{}
var _ suite.TearDownAllSuite = ValidSuite{}
var _ suite.TearDownTestSuite = ValidSuite{}
var _ suite.BeforeTest = ValidSuite{}
var _ suite.AfterTest = ValidSuite{}
var _ suite.WithStats = ValidSuite{}
var _ suite.SetupSubTest = ValidSuite{}
var _ suite.TearDownSubTest = ValidSuite{}
```

Reply:
👍 done